agent/internal/binding-accessor/binding-accessor.go

```diff
@@ -33,10 +34,19 @@ type BindingAccessorFunc func(ctx Context) (value interface{}, err error)
 // be combined together in order to represent a whole binding accessor
 // expression through different function closures (field access, array access,
 // map access, etc.).
-type valueFunc func(ctx Context) (value interface{})
+type valueFunc func(ctx Context, maxDepth int) (value interface{})
```

Comment:
Good thing to do is to have a `type Value interface{}` and use that everywhere in there, casting it to `interface{}` only if/when necessary.

Reply:
It wouldn't prevent from it (cf. https://play.golang.org/p/0ZwcvOYR1CW).
To do what you suggest, you need to add a method in the interface such as `type Value interface{ isValue() }` to enforce a given list of types implementing it.

For now that I don't do JS callback yet, I feel like it is safe enough to allow every Go value result and let the result consumer do the type checks (the WAF value marshaler returns an error for example).